corehq/apps/export/models/new.py

```diff
@@ -2470,6 +2477,48 @@ def user_to_be_notified(cls, domain, user):
             )
         )
 
+
+class DataFile(models.Model):
+    domain = models.CharField(max_length=126, db_index=True)
+    filename = models.CharField(max_length=255)
+    description = models.CharField(max_length=255)
+    content_type = models.CharField(max_length=255)
+    blob_id = models.CharField(max_length=255)
+    content_length = models.IntegerField(null=True)
+
+    class Meta(object):
+        app_label = 'export'
+
+    def get_blob(self):
+        db = get_blob_db()
+        try:
+            blob = db.get(self.blob_id)
+        except (KeyError, NotFound) as err:
+            raise NotFound(str(err))
+        return blob
+
+    def save_blob(self, uploaded_file):
+        content_string = StringIO()
+        for chunk in uploaded_file.chunks():
+            content_string.write(chunk)
+        self.content_type = magic.from_buffer(content_string.read(1024))
```

Comment:
The content type that we get from the request is whatever the user's browser tells us it is. I'm going with "trust but verify" here by checking the file ourselves and updating the value. But you make an important point -- if we don't use the value, does it matter what the content type is? I'm undecided but I'll look up what can go wrong, and consider the fact that at the moment only admins can upload files. It will be nice to be able to take this out.